src/PrestaShopBundle/Controller/Admin/Improve/International/TaxRulesGroupController.php

```diff
@@ -0,0 +1,343 @@
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_tax_rules_groups_index",
+     * )
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        return $this->redirect($this->getAdminLink('AdminTaxRulesGroup', []));
+    }
+
+    /**
+     * Handles tax rules group edit
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_tax_rules_groups_index",
+     * )
+     *
+     * @param Request $request
+     * @param int $taxRulesGroupId
+     *
+     * @return Response
+     */
+    public function editAction(Request $request, int $taxRulesGroupId): Response
+    {
+        return $this->redirect($this->getAdminLink('AdminTaxRulesGroup', []));
+    }
+
+    /**
+     * Deletes tax rules group.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_tax_rules_groups_index",
+     * )
+     *
+     * @param int $taxRulesGroupId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction(int $taxRulesGroupId): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteTaxRulesGroupCommand((int) $taxRulesGroupId));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion.', 'Admin.Notifications.Success')
+            );
+        } catch (TaxRulesGroupException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->redirectToRoute('admin_tax_rules_groups_index');
+    }
+
+    /**
+     * Toggles status.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_tax_rules_groups_index",
+     * )
+     *
+     * @param int $taxRulesGroupId
+     *
+     * @return RedirectResponse
+     */
+    public function toggleStatusAction(int $taxRulesGroupId): RedirectResponse
+    {
+        try {
+            /** @var EditableTaxRulesGroup $editableTaxRulesGroup */
+            $editableTaxRulesGroup = $this->getQueryBus()->handle(
+                new GetTaxRulesGroupForEditing((int) $taxRulesGroupId)
+            );
+
+            $this->getCommandBus()->handle(
+                new ToggleTaxRulesGroupStatusCommand((int) $taxRulesGroupId, !$editableTaxRulesGroup->isActive())
+            );
+
+            $this->addFlash(
+                'success',
+                $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success')
+            );
+        } catch (TaxRulesGroupException $e) {
```

Comment:
We must catch **all** Exceptions here because we want to be able to display nicely ModuleErrorExceptions
see https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Module/Exception/ModuleErrorException.php

Reply:
```suggestion
        } catch (\Exception $e) {
```
=> do this everywhere 😉